ray-operator/hack/update-codegen.sh

```diff
@@ -23,44 +23,19 @@ echo ">> Using ${CODEGEN_PKG}"
 
 cd "${SCRIPT_ROOT}"
 
-# Migrate to using kube_codegen.sh once the following issue is fixed:
-# https://github.com/kubernetes/code-generator/issues/165
-
-go install "${CODEGEN_PKG}"/cmd/applyconfiguration-gen
-go install "${CODEGEN_PKG}"/cmd/client-gen
-go install "${CODEGEN_PKG}"/cmd/lister-gen
-go install "${CODEGEN_PKG}"/cmd/informer-gen
-
-"${GOPATH}"/bin/applyconfiguration-gen \
-  --input-dirs github.com/ray-project/kuberay/ray-operator/apis/ray/v1 \
-  --external-applyconfigurations k8s.io/api/core/v1.PodTemplateSpec:k8s.io/client-go/applyconfigurations/core/v1 \
-  --output-package github.com/ray-project/kuberay/ray-operator/pkg/client/applyconfiguration \
-  --go-header-file hack/boilerplate.go.txt \
-  --output-base "${SCRIPT_ROOT}" \
-  --trim-path-prefix ${ROOT_PKG}
-
-"${GOPATH}"/bin/client-gen \
-  --input github.com/ray-project/kuberay/ray-operator/apis/ray/v1 \
-  --input-base="" \
-  --apply-configuration-package=github.com/ray-project/kuberay/ray-operator/pkg/client/applyconfiguration \
-  --clientset-name "versioned"  \
-  --output-package github.com/ray-project/kuberay/ray-operator/pkg/client/clientset \
-  --go-header-file hack/boilerplate.go.txt \
-  --output-base "${SCRIPT_ROOT}" \
-  --trim-path-prefix ${ROOT_PKG}
-
-"${GOPATH}"/bin/lister-gen \
-  --input-dirs github.com/ray-project/kuberay/ray-operator/apis/ray/v1 \
-  --output-package github.com/ray-project/kuberay/ray-operator/pkg/client/listers \
-  --go-header-file hack/boilerplate.go.txt \
-  --output-base "${SCRIPT_ROOT}" \
-  --trim-path-prefix ${ROOT_PKG}
-
-"${GOPATH}"/bin/informer-gen \
-  --input-dirs github.com/ray-project/kuberay/ray-operator/apis/ray/v1 \
-  --versioned-clientset-package github.com/ray-project/kuberay/ray-operator/pkg/client/clientset/versioned \
-  --listers-package github.com/ray-project/kuberay/ray-operator/pkg/client/listers \
-  --output-package github.com/ray-project/kuberay/ray-operator/pkg/client/informers \
-  --go-header-file hack/boilerplate.go.txt \
-  --output-base "${SCRIPT_ROOT}" \
-  --trim-path-prefix ${ROOT_PKG}
+# shellcheck source=/dev/null
+source "${CODEGEN_PKG}/kube_codegen.sh"
+
+# Generating conversion and defaults functions
+kube::codegen::gen_helpers \
+  --boilerplate "${SCRIPT_ROOT}/hack/boilerplate.go.txt" \
+  "${SCRIPT_ROOT}/apis"
+
+kube::codegen::gen_client \
+  --boilerplate "${SCRIPT_ROOT}/hack/boilerplate.go.txt" \
+  --output-dir "${SCRIPT_ROOT}/pkg/client" \
+  --output-pkg "${ROOT_PKG}/pkg/client" \
+  --with-watch \
+  --with-applyconfig \
+  --applyconfig-externals k8s.io/api/core/v1.PodTemplateSpec:k8s.io/client-go/applyconfigurations/core/v1 \
```

Comment:
```suggestion
  --applyconfig-externals k8s.io/api/core/v1.PodTemplateSpec:k8s.io/client-go/applyconfigurations/core/v1 \
  --one-input-api ray/v1 \
```
Should add this or `ray-operator/apis/ray/v1alpha1` will also be seen as a target.

Reply:
It seems they are both seen now after this: https://github.com/ray-project/kuberay/pull/3004#discussion_r1966454087

Reply:
great advice, thanks